spec/requests/alert_definitions_spec.rb

```diff
@@ -306,4 +306,24 @@
     expect(alert_definition_profiles.first.reload.description).to eq("Updated Test Alert Profile 1")
     expect(alert_definition_profiles.second.reload.description).to eq("Updated Test Alert Profile 2")
   end
+
+  it "assings an alert definition to a profile" do
+    api_basic_authorize
+    alert_definition_profile = FactoryGirl.create(:miq_alert_set)
+    alert_definition = FactoryGirl.create(:miq_alert)
+    post(api_alert_definition_profile_url(nil, alert_definition_profile) + "/alert_definitions", :params => gen_request(:assign, alert_definition))
+    expect(response).to have_http_status(:ok)
+    expect(response.parsed_body["results"].count).to eq(1)
+    expect(alert_definition_profile.reload.members.first).to eq(alert_definition)
```

Comment:
Prefer to make this (and other) expectations on the response itself. Here, you're testing something about the implementation, which belongs in a lower level test. Request specs should be more concerned with how this behavior is revealed through the response - perhaps you can examine the contents of `"results"` and put some expectations there instead?

Reply:
I changed the tests accordingly, and you were right - this helped me find a bug in the unassign method (which I now fixed).